quinn-proto/src/connection/mod.rs

```diff
@@ -362,6 +362,20 @@ where
             return None;
         }
 
+        let mut num_datagrams = 0;
+        // TODO: Check whether this can be constant or there are situations where
+        // we have to enforce less packets due to path limitations. Seems like
+        // we are beyond anti-amplification check this might already be good.
```

Comment:
The existing anti-amplification check assumes that we're sending at most one packet at a time, because it checks that we're at least one MTU away from violating the limit. To preserve it with this change, we could increment `self.total_sent` and re-check `self.anti_amplification_blocked()` after recording each packet into the GSO buffer before beginning the next.